contracts/src/staking/BeaconDepositContract.sol

```diff
@@ -80,7 +100,13 @@ contract BeaconDepositContract is IBeaconDepositContract, Ownable {
             revert InvalidSignatureLength();
         }
 
-        uint64 amountInGwei = _deposit(amount);
+        // Update the pubkey to operator mapping if first deposit.
+        if (_pubkeyToOperator[pubkey] == address(0)) {
+            _pubkeyToOperator[pubkey] = msg.sender;
```

Comment:
i feel like this would be more extensible if we just allowed for operator to be passed in as an arg.

Reply:
that will just make operator change process one step, i feel it should be two step process for validators to change their operator address, its better security wise.

Reply:
one step is fine imo they already comitting capital so this should be verified before hand